slang_frontend.cc

```diff
@@ -982,20 +979,20 @@ struct ProceduralVisitor : public ast::ASTVisitor<ProceduralVisitor, true, false
 		for (auto init : stmt.initializers)
 			eval(*init);
 
-		disableLoop_ins = {};
-		disableLoop_outs = {};
-
-
 		int ncycles = 0;
 
 		if (!stmt.stopExpr) {
 			diag_scope->addDiag(diag::MissingStopCondition, stmt.sourceRange.start());
 			return;
 		}
-
 		SwitchHelper b(current_case, vstate, {RTLIL::S0});
 		b.sw->statement = &stmt;
```

Comment:
The SwitchHelper needs to be constructed once per iteration, we cannot reuse it

Reply:
Ah sorry, so then we are fully discarding the use of a switch/case?

Reply:
Nvm, I get it

Reply:
Pretty much, or rather we need to have a new switch per iteration (like you had it originally)